core/src/com/ericc/the/game/systems/logic/FieldOfViewSystem.java

```diff
@@ -0,0 +1,204 @@
+import java.util.Arrays;
+
+public class FieldOfViewSystem extends EntitySystem {
+
+    private Map map;
+    private ImmutableArray<Entity> otherMobs;
+    private ImmutableArray<Entity> heroes;
+    private ScreenBoundsGetter sbh;
+
+    // a helper data structure with possible moves from one tile in horizontal and vertical directions
+    private static ArrayList<GridPoint2> moves =
+            new ArrayList<>(Arrays.asList(
+                    new GridPoint2(1, 0),
+                    new GridPoint2(0, 1),
+                    new GridPoint2(-1, 0),
+                    new GridPoint2(0, -1)
+            ));
+
+    // stores moves on diagonals
+    private static ArrayList<GridPoint2> cornerMoves =
+            new ArrayList<>(Arrays.asList(
+                    new GridPoint2(1, 1),
+                    new GridPoint2(-1, -1),
+                    new GridPoint2(-1, 1),
+                    new GridPoint2(1, -1)
+            ));
+
+    private static ArrayList<ArrayList<GridPoint2>> corners =
+            new ArrayList<>(
+                    Arrays.asList(
+                            new ArrayList<>(
+                                    Arrays.asList(
+                                            new GridPoint2(-1, 0),
+                                            new GridPoint2(0, 1)
+                                    )
+                            ),
+                            new ArrayList<>(
+                                    Arrays.asList(
+                                            new GridPoint2(0, 1),
+                                            new GridPoint2(1, 0)
+                                    )
+                            ),
+                            new ArrayList<>(
+                                    Arrays.asList(
+                                            new GridPoint2(1, 0),
+                                            new GridPoint2(0, -1)
+                                    )
+                            ),
+                            new ArrayList<>(
+                                    Arrays.asList(
+                                            new GridPoint2(0, -1),
+                                            new GridPoint2(-1, 0)
+                                    )
+                            )
+                    )
+            );
+
+    public FieldOfViewSystem(Map map, ScreenBoundsGetter sbh) {
+        this.map = map;
+        this.sbh = sbh;
+    }
+
+    @Override
+    public void addedToEngine(Engine engine) {
+        otherMobs = engine.getEntitiesFor(
+                Family.all(PositionComponent.class, FieldOfViewComponent.class).exclude(PlayerComponent.class).get()
+        );
+        heroes = engine.getEntitiesFor(
+                Family.all(PositionComponent.class, FieldOfViewComponent.class, PlayerComponent.class).get()
+        );
+    }
+
+    @Override
+    public void update(float deltaTime) {
+        sbh.update();
+
+        for (Entity entity : heroes) {
```

Comment:
po co jest w ogóle to rozróżnienie na heroes i other skoro dla jednych i drugich robisz to samo?

Reply:
z perspektywy systemu to nie ma znaczenia (i raczej nie powinno mieć) czy to jest hero czy nie